components/searchable_user_list/searchable_user_list.jsx

```diff
@@ -172,10 +172,14 @@ class SearchableUserList extends React.PureComponent {
             usersToDisplay = this.props.users;
         } else if (!this.props.term) {
             const pageStart = this.props.page * this.props.usersPerPage;
-            const pageEnd = pageStart + this.props.usersPerPage;
+            let pageEnd = pageStart + this.props.usersPerPage;
+            if (this.props.users.length < pageEnd) {
+                pageEnd = this.props.users.length;
+            }
+
             usersToDisplay = this.props.users.slice(pageStart, pageEnd);
 
-            if (pageEnd < this.props.users.length) {
+            if (pageEnd < this.props.total) {
```

Comment:
For context on this fix, `this.props.users` is all currently fetched users on this team, with the deleted users filtered out client-side, and `this.props.total` is the total number of non-deleted users on this team, taken from the `stats` API endpoint.